flow/record/fieldtypes/__init__.py

```diff
@@ -734,3 +735,98 @@ def __repr__(self) -> str:
                 quote = '"'
 
         return f"{quote}{s}{quote}"
+
+
+class command(FieldType):
+    executable: Optional[path] = None
+    args: Optional[list[str]] = None
+
+    _path_type: type[path] = None
+    _posix: bool
+
+    def __new__(cls, value: str) -> command:
+        if cls is not command:
+            return super().__new__(cls)
+
+        if not isinstance(value, str):
+            raise ValueError(f"Expected a value of type 'str' not {type(value)}")
+
+        # pre checking for windows like paths
+        # This checks for windows like starts of a path:
+        #   an '%' for an environment variable
+        #   r'\\' for a UNC path
+        #   the strip and check for ":" on the second line is for `<drive_letter>:`
+        windows = value.startswith((r"\\", "%")) or value.lstrip("\"'")[1] == ":"
+
+        if windows:
+            cls = windows_command
+        else:
+            cls = posix_command
+        return super().__new__(cls)
+
+    def __init__(self, value: str | tuple):
+        if value is None:
+            return
+
+        if isinstance(value, tuple):
+            executable, self.args = value
+            self.executable = self._path_type(executable)
+            self.args = list(self.args)
+            return
+
+        self.executable, self.args = self._split_function(value)
```

Comment:
Maybe better to swap this around? `if isinstance(value, str)` and fall through for the tuple? That way it can also work for a list.